### PR TITLE
improve say command + message when command not found

### PR DESCRIPTION
**Pull request by qexat**
fix issues #72 & #73

### CODE REVIEW

File: src/commands/fun/say.shaii.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default defineCommand({
   requiresProcessing: false,
   execute: async (message) => {
     if (message.args.length === 0) return `what do you want to say ${randomChoice(SLURS)}`;
+	await message.delete();
```
Comment (Geoxor) [inline src/commands/fun/say.shaii.ts]: ## Don't await the delete 
We do not care if the message is deleted or not and we don't wanna block the thread till this deletes
```ts
message.delete().catch(() => {})
```

File: src/shaii/Shaii.shaii.ts [hunk 2]
```diff
@@ -227,8 +228,25 @@ class Shaii {
...
           // If it doesn't exist we respond
-          if (!command)
-            return message.reply(`That command doesn't exist ${randomChoice(SLURS)}`).catch(() => {});
+          if (!command) {
+            let closestCommand = {
+              name: "",
+              ld: 4, // ld: levenshtein distance
+            };
+            for (const [command_name] of this.commands.entries()) {
+              var currentCommandLD = levenshtein(command_name, message.command);
```
Comment (Geoxor) [inline src/shaii/Shaii.shaii.ts:237]: use `const`

File: src/shaii/Shaii.shaii.ts [hunk 3]
```diff
@@ -227,8 +228,25 @@ class Shaii {
...
+
+              if (currentCommandLD < closestCommand.ld)
+                (closestCommand.name = command_name), (closestCommand.ld = currentCommandLD);
+            }
+            var suggestion: string;
```
Comment (Geoxor) [inline src/shaii/Shaii.shaii.ts:242]: use `let`

**Review by Geoxor — CHANGES_REQUESTED** [review]

File: src/commands/fun/say.shaii.ts (hunk 1 above)
Comment (qexat) [reply src/commands/fun/say.shaii.ts]: I changed it, but that means we have to find a solution to make Shaii sending the result without replying to the deleted message

**Review by Geoxor — APPROVED** [review]

**Pull request merged**